lab-yunjoo/app/component/list/index.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+require('./list.scss');
+const angular = require('angular');
+angular.module('demoApp').directive('appList', function(){
+  return {
+    restrict: 'E',
+    replace: true,
+    template: require('./list.html'),
+    controller: ['$log','listService','noteService',ListController],
+    controllerAs: 'listCtrl',
+    bindToController: true,
+    scope: {
+      list: '='
+    }
+  };
+});
+
+function ListController($log, listService, noteService){
+  this.deleteList = function(){
+    $log.debug('listCtrl.deleteList');
+    listService.deleteList(this.list._id)
+    .catch(() => {
+      alert('you failed to deleteme yanoob');
+    });
+  };
+
+  this.createNote = function(data){
+    $log.debug('listCtrl.createNote');
+    data.listId = this.list._id;
+
+    noteService.createNote(data)
+    .then( note => {
+      this.list.notes.push(note);
+    })
+    .catch( () => {
+      alert('no no createNote');
```

Comment:
It's usually a good idea to take an error as an argument in catch blocks so that you can log it and see what it was.